sdk/persist_test.go

```diff
@@ -0,0 +1,86 @@
+package sdk
+
+import (
+	"bytes"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+// Tests may fail if run on Windows, unix permissioning
```

Comment:
Will permissions setting not work on windows at all? I think we're trying somewhat to support windows but unknown if things would currently work.

Reply:
Looks like Chown fails on windows. Wonder if its enough to just Chmod. Likely if some other user owns the directories then attempting to Chown would fail anyway unless we can CLI with sudo/root.

Reply:
Okay, I went through a bunch of convoluted scenarios and I think you're right, it's okay to remove the chown. By ensuring that we are able to change the permissions even if someone else creates and owns the file we can block them from accessing it after write

Reply:
Sounds good! Thanks for digging in.